### ISSUE TITLE
[Design-Issue:SDK] Why two Estimators (Estimator vs. SKLearn) for running Scikit-Learn models if there's no additional benefit/advantage for any of them?

### ISSUE THREAD

**Issue by CESARDELATORRE** [danielsc/azureml-workshop-2019#26]
Afaik, it is almost the same to use the plain Estimator or the SKLearn estimator with no additional benefit or reasons why you should use one or the other. If that's confirmed, why do we have two approaches for doing the same thing? One of the two APIs should be deprecated:

```
# Using Estimator class
estimator = Estimator(source_directory=project_folder, 
                      script_params=script_params,
                      compute_target=cluster,
                      entry_script='train.py',
                      pip_packages=pip_packages,
                      conda_packages=['scikit-learn==0.20.3'],
                      inputs=[ws.datasets['IBM-Employee-Attrition'].as_named_input('attrition')])


# Using SKLearn estimator class
estimator = SKLearn(source_directory=project_folder, 
                     script_params=script_params,
                     compute_target=cluster,
                     entry_script='train.py',
                     pip_packages=pip_packages,
                     conda_packages=['scikit-learn==0.20.3'],
                     inputs=[aml_dataset.as_named_input('attrition')])
```

**Comment by danielsc** [comment]
clearly a regression: https://msdata.visualstudio.com/Vienna/_workitems/edit/583743

**Comment by CESARDELATORRE** [comment]
File explicit bug/papercut. The regression bug above doesn't look the same design issue. 

https://msdata.visualstudio.com/Vienna/_workitems/edit/587848

**Comment by diondrapeck** [comment]
Hi @CESARDELATORRE, @danielsc - the two offerings are not the same, and there isn't a regression. 

Framework estimators add their additional libraries by setting a framework-specific Docker image as the estimator base image - not by adding them via pip_packages or conda_packages**. Hence why you won't find them by calling 

`print(estimator.conda_dependencies.serialize_to_string())`

However, you can see the that the sklearn base image is being used by running:

`estimator.run_config.environment.docker.base_image`.

For SKLearn estimator, that will output: "sklearn:0.20.3-cpu". The information of which packages are included in this image is listed in [the class documentation](https://docs.microsoft.com/en-us/python/api/azureml-train-core/azureml.train.sklearn.sklearn?view=azure-ml-py). This is true for all framework estimators. 

If you use the SKLearn estimator class to run an experiment with a script that imports the scikit-learn package - without adding scikit-learn to pip_packages or conda_packages - your experiment will run as expected. Of course, if you tried this with the generic estimator, it would fail.


** There is one set of circumstances where packages would be added using conda_dependencies, but it's not worth getting into here because it's technical and does not ultimately affect the function of the object
